kedro/io/data_catalog.py

```diff
@@ -170,7 +203,15 @@ def __init__(
         self._data_sets = dict(data_sets or {})
         self.datasets = _FrozenDatasets(self._data_sets)
         self.layers = layers
-
+        # Keep a record of all patterns in the catalog.
+        # {dataset pattern name : dataset pattern body}
+        self.dataset_patterns = dict(dataset_patterns or {})
+        # Sort all the patterns according to the parsing rules -
+        self._sorted_dataset_patterns = _sort_dataset_factory_patterns(
+            list(self.dataset_patterns.keys())
+        )
+        # Cache that stores {name : matched_pattern}
+        self._pattern_matches_cache: dict[str, str] = {}
```

Comment:
Do we really need a separate variable for this cache? Couldn't this just be `self._data_sets` and just have a `set()` of all original (non-pattern) data set names?